ansible/templates/minigraph_cpg.j2

```diff
@@ -197,6 +204,52 @@
         <a:RouteMaps/>
       </a:BGPRouterDeclaration>
 {% endfor %}
+{% if switch_type is defined and switch_type == 'voq' %}
+{% for a_linecard in all_inbands %}
+{% if a_linecard != inventory_hostname %}
```

Comment:
how about iBGP peers within a linecard that has multiple asics ?

Reply:
all_inbands now is a dictionary with key being each linecard and value being a list of inband ip's. For 2 linecards (board3 and board4) with 2 asics, it would look like:

```
 "all_inbands": {
            "board3": [
                "3.3.3.4/32",
                "3.3.3.5/32"
            ],
            "board4": [
                "3.3.3.10/32",
                "3.3.3.11/32"
            ]
        }
```

And the voq_peers generated above has key for all the inbands across all linecards and its correspoding router name. So, for the above example when running through board3, it would be:

```
{
   '3.3.3.4' : 'ASIC0',
   '3.3.3.5':  'ASIC1',
   '3.3.3.10': 'board4-ASIC0'
   '3.3.3.11': 'board4-ASIC1'
}
```

Reply:
a_linecard != inventory_hostname check will not add the iBGP between 3.3.3.4 and 3.3.3.5 when running through board3 right ?

Reply:
Here we are defining BgpRouterDeclaration for all other asics in all other linecards - not iBGP sessions. That is taken care of above in lines 82 - 115.